packages/hawtio/src/preferences/HomePreferences.tsx

```diff
@@ -1,13 +1,46 @@
 import { Button, CardBody, Form, FormGroup, FormSection, Switch } from '@patternfly/react-core'
 import React, { useState } from 'react'
-import { log } from './globals'
+
+type LocalStorageFieldData<T> = {
+  localStorageKey: string
+  defaultValue: T
+  updateFunction: React.Dispatch<React.SetStateAction<T>>
+}
 
 export const HomePreferences: React.FunctionComponent = () => {
-  const [defaultVerticalNavState, setDefaultVerticalNavState] = useState(true)
+  const LOCAL_STORAGE_VERTICAL_NAVIGATION = 'local_storage_vertical_navigation'
+  const DEFAULT_VALUE_VERTICAL_NAVIGATION = true
+
+  const VERTICAL_NAVIGATION_INITIAL_VALUE: boolean =
+    localStorage.getItem(LOCAL_STORAGE_VERTICAL_NAVIGATION) !== null
+      ? localStorage.getItem(LOCAL_STORAGE_VERTICAL_NAVIGATION) === 'true'
+      : DEFAULT_VALUE_VERTICAL_NAVIGATION
```

Comment:
I didn't find a clever way to put this. I need to first check if it's null to know if I need to use the DEFAULT_VERTICAL_NAVIGATION.

Then, on top of that, TS doesn't have a Boolean.parseBoolean like Java, so it ends being a bit ugly :D

Reply:
You can always look at the code in the other places that interact with local storage.